Public/Src/Engine/Scheduler/FileSystem/FileSystemView.cs

```diff
@@ -151,7 +151,9 @@ public Possible<PathExistence> TryEnumerateDirectory(AbsolutePath path, FileSyst
             PathExistence existence;
             if (PathExistenceCache.TryGetValue(path, out entry) && entry.TryGetExistence(mode, out existence))
             {
-                if (existence != PathExistence.ExistsAsDirectory)
+                if (existence == PathExistence.Nonexistent

```

Comment:
>Nonexistent [](start = 47, length = 11)

Why do we do this for Nonexistent as well?

Reply:
I'm preserving the old behavior where if it is Nonexistent, then we return it immediately.

---
In reply to: [283897412](https://github.com/microsoft/BuildXL/pull/301#discussion_r283897412) [](ancestors = 283897412)